### ISSUE TITLE
IconOnly Buttons always has a onhover tooltip

### ISSUE THREAD

**Issue by nicoxxie** [carbon-design-system/carbon#5069]
## What package(s) are you using?

- [x] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

> Describe in detail the issue you're having.
IconOnly Buttons always has a tooltip even with no text, and our design does not have a tooltip for this button.

![Screen Shot 2020-01-15 at 2 38 51 PM](https://user-images.githubusercontent.com/17915105/72541629-77630400-3848-11ea-82b0-a6f6e1a18659.png)

```
const assistiveText = hasIconOnly ? (
    <span className={`${prefix}--assistive-text`}>{iconDescription}</span>
  ) 
```
> Is this issue related to a specific component?
Button

> What did you expect to happen? What happened instead? What would you like to
> see changed?
support not showing tooltip on hover

> What browser are you working in?
Chrome Version 79.0.3945.88 (Official Build) (64-bit)

> What version of the Carbon Design System are you using?
carbon-components-react: "7.9.0"
carbon-components: "10.9.0"

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?
IBM Cloud

## Steps to reproduce the issue

1. Step one
2. Step two
3. Step three
4. etc.

> Please create a reduced test case in CodeSandbox
>
> - Style and vanilla JS:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/components/examples/codesandbox
> - React:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/react/examples/codesandbox

## Additional information

- Screenshots or code
- Notes

**Comment by tw15egan** [comment]
I believe this is an accessibility requirement

cc @dakahn

**Comment by emyarod** [comment]
yeah that was the intention behind coupling the tooltip with the icon only button

ref (https://github.com/carbon-design-system/carbon/issues/2293 https://github.com/carbon-design-system/carbon/pull/2366)

**Comment by dakahn** [comment]
The tooltip is an accessibility affordance for sighted users. Putting 100% faith in your iconography to convey a button's action is risky -- even if the icon can seem obvious. Try succinctly describing whatever that button's action is in the tooltip. Some examples for that icon could be: 

"Search for {whatever is being queried}"
"Search for {whatever is being queried} filtered by {existing filters}"

**Comment by qkrooz** [comment]
Solve with <Button hasIconOnly style={{overflow:'hidden'}}></Button>
